src/ess/polarization/correction.py

```diff
@@ -0,0 +1,168 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import sciline
+import scipp as sc
+
+from .types import (
+    Analyzer,
+    AnalyzerSpin,
+    Down,
+    PolarizationCorrectedData,
+    PolarizationCorrection,
+    Polarizer,
+    PolarizerSpin,
+    PolarizingElement,
+    PolarizingElementCorrection,
+    ReducedSampleDataBySpinChannel,
+    TransmissionFunction,
+    Up,
+)
+
+
+def compute_polarizing_element_correction(
+    channel: ReducedSampleDataBySpinChannel[PolarizerSpin, AnalyzerSpin],
+    transmission: TransmissionFunction[PolarizingElement],
+) -> PolarizingElementCorrection[PolarizerSpin, AnalyzerSpin, PolarizingElement]:
+    """
+    Compute matrix coefficients for the correction of a polarizing element.
+
+    The coefficients stem from the inverse of a symmetric matrix of the form
+    [[Tplus, Tminus], [Tminus, Tplus]]. The inverse is given by a matrix
+        mat = 1/denom * [[Tplus, -Tminus], [-Tminus, Tplus]],
+    with
+        denom = Tplus**2 - Tminus**2.
+    As there are only two unique elements in the matrix, we return them as a dataclass
+    with diagonal and off-diagonal elements.
+
+    Parameters
+    ----------
+    channel :
+        Data including wavelength (and time) for a given spin channel.
+    transmission :
+        Transmission function for the polarizing element.
+
+    Returns
+    -------
+    :
+        Correction matrix coefficients.
+
+    """
+    t_plus = transmission.apply(channel, 'plus')
```

Comment:
Why did you move away from using `__call__` here? Seems odd to create a new interface for something that is very similar to a regular function call.

Reply:
I left `__call__` as a regular interface, taking the required params, e.g., for fitting. `applu` is a wrapper extracting the required coords.